### PR TITLE
Refactor SDEProblem constructor

### PR DESCRIPTION
**Pull request by ErikQQY**
Now the SDE constructor is simply ```SDEProblem(SDEFunction(f, g), u0, tspan)``` or ```SDEProblem(f, g, u0, tspan)```

https://github.com/SciML/SciMLBase.jl/pull/489

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This PR seems to fail a lot of tests.

**Comment by ErikQQY** [comment]
Now these failings seem caused by SDEProblemLibrary.jl

**Comment by ErikQQY** [comment]
It seems the testing for StochasticDiffEq.jl is still using the old version of SDEProblemLibrary.jl, is there some way we can bump it?

**Comment by ChrisRackauckas** [comment]
Up the lower bound.

**Comment by ErikQQY** [comment]
The lower bound of SDEProblemLibrary.jl in tests? I don't think there is one.

**Comment by ChrisRackauckas** [comment]
Looks like it got the updated SDEProblemLibrary

**Comment by ErikQQY** [comment]
Yeah, it is using the latest SDEProblemLibrary, but the error is still about Catalyst and MTK stuff, is there something I am missing?

**Comment by ChrisRackauckas** [comment]
No, those test errors are real: https://github.com/SciML/StochasticDiffEq.jl/actions/runs/6291363683/job/17079673649?pr=546#step:6:548

Was the tests all passing locally? That seems unlikely.

File: Project.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ OrdinaryDiffEq = "6.52"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "2.0.1"
```
Comment (ErikQQY) [inline Project.toml:52]:
```suggestion
SciMLBase = "2.0.2"
```

**Comment by ErikQQY** [comment]
The errors just remind me that the change of ```SDEProblem``` constructor also affects ```SplitSDEProblem``` and etc., may need a little longer to get all this fixed. See:
https://github.com/SciML/StochasticDiffEq.jl/blob/0e396bfa4d1a2a9bf4a2a2be9ea79842f8c4a5a3/test/iif_methods.jl#L2-L15

File: Project.toml [hunk 2]
```diff
@@ -49,7 +49,7 @@ OrdinaryDiffEq = "6.52"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "2.0.2"
```
Comment (ErikQQY) [inline Project.toml:52]:
```suggestion
SciMLBase = "2.0.3"
```

**Comment by ChrisRackauckas** [comment]
Did you forget to push an overload to `remake` like https://github.com/SciML/SciMLBase.jl/blob/master/src/remake.jl#L52 ?

**Comment by ErikQQY** [comment]
Yeah, I need to make a dispatch of ```remake``` for ```SDEProblem```

File: Project.toml [hunk 3]
```diff
@@ -49,7 +49,7 @@ OrdinaryDiffEq = "6.52"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "2.0.3"
```
Comment (ErikQQY) [inline Project.toml:52]:
```suggestion
SciMLBase = "2.0.4"
```

**Comment by ErikQQY** [comment]
Tests failings are real, still need some modifications on ```SplitSDEProblem``` and ```DynamicalSDEProblem```

File: Project.toml [hunk 4]
```diff
@@ -49,7 +49,7 @@ OrdinaryDiffEq = "6.52"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "2.0.4"
```
Comment (ChrisRackauckas) [inline Project.toml:52]:
```suggestion
SciMLBase = "2.0.5"
```

File: Project.toml [hunk 5]
```diff
@@ -49,7 +49,7 @@ OrdinaryDiffEq = "6.52"
 RandomNumbers = "1.5.3"
 RecursiveArrayTools = "2"
 Reexport = "0.2, 1.0"
-SciMLBase = "1.70"
+SciMLBase = "2.0.5"
```
Comment (ChrisRackauckas) [inline Project.toml:52]:
```suggestion
SciMLBase = "2.0.6"
```

**Comment by ErikQQY** [comment]
Finally all the issues are addressed🎉🎉🎉

**Comment by ChrisRackauckas** [comment]
* https://buildkite.com/julialang/stochasticdiffeq-dot-jl/builds/543#0189b12e-5acc-417f-9687-7b7dea3a4354
* https://github.com/SciML/StochasticDiffEq.jl/actions/runs/6350081193/job/17249285877?pr=546

Why are these slower by hours?

**Comment by ChrisRackauckas** [comment]
Interesting, that looks like it was just a random hiccup.

**Comment by prbzrg** [comment]
I have a suggestion:
It would be great if we had [`PkgBenchmark`](https://github.com/JuliaCI/PkgBenchmark.jl) setup, we could have [`BenchmarkCI`](https://github.com/tkf/BenchmarkCI.jl) to generate a report for each PR. improving performance with an independent report is more convenient. And we would be sure that PRs aren't detrimental to performance.
I am interested to create it, but I don't know where to start.

**Comment by ChrisRackauckas** [comment]
It looks like DiffEqNoiseProcess's regression is on master, so this can be merged. https://github.com/SciML/StochasticDiffEq.jl/actions/runs/6353130403/job/17257204276?pr=549

**Pull request merged**